### DISCUSSION TITLE
empty array again - how to pass props properly to pages?

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#257] [Web] [answered]
Hallo zusammen,
ich habe meine App nun um zwei Seiten erweitert und muss deshalb meine States auf die `_app.js` verlagern. Beim Runterreichen an die pages kommt `timerChain` dann aber wieder nur als leeres array an... 

Den `useEffect` vom useLocalStorage-Problem jetzt auch auf der `timer.js` einzubauen hat nicht geholfen:
```
useEffect(() => {
    setNewTimerChain(timerChain);
  }, [timerChain]);
```

Kann sich das jemand bitte anschauen?
Vielen Dank! :)

[Pullrequest](https://github.com/DanielGennat/capstone-project/pull/22)

**Comment by ThomasKirst** [comment]
Hi Daniel,
in your `_app.js` you need to pass the correct property for the `timerPointer`:

```diff
+++ b/pages/_app.js
@@ -31,7 +31,7 @@ function MyApp({ Component, pageProps }) {
         {...pageProps}
         timerChain={timerChain}
         setTimerChain={setTimerChain}
-        timerPointer={setTimerPointer}
+        timerPointer={timerPointer}
         countdownKey={countdownKey}
         setCountdownKey={setCountdownKey}
       />
```

It needs to be a numeric value instead of a setter function ;)

**Reply by DanielGennat** [reply]
oh no! :) coding after 10 pm :innocent:
Thanks! :)

**Comment by DanielGennat** [comment, marked as answer]
Hallo Thomas,
die timer page funktioniert jetzt wieder. Allerdings werden die Änderungen vom Formular (jetzt edit page) `onSubmit` nicht mehr übernommen / hochgereicht.
Kannst du dir das bitte nochmal anschauen?

https://github.com/DanielGennat/capstone-project/pull/22

Vielen Dank :)

**Reply by DanielGennat** [reply]
Okay, I think I fixed it :)

In addition to Thoma' answer I had to use `useLocalStorage` for the `newTimerChain`State and delete the `useEffect` :
```
//edit.js

  const [newTimerChain, setNewTimerChain] = useLocalStorage(
    '_timerChainStorage',
    timerChain
  );

  // useEffect(() => {
  //   setNewTimerChain(timerChain);
  // }, [timerChain]);
```
